### ISSUE TITLE
Make it possible to define target class and ordinal

### ISSUE THREAD

**Issue by OndrejM** [eclipse/microprofile-config#126]
Since when passing a lambda to `builder.withConverters` the return type of the lambda doesn't contain generic type, it would make sense to create an additional method for a single converter, with a second argument to specify the converter's type - the builder pattern is aimed at power users and they may want to specify exact converter type anyway.

It would be also suitable, if a user would want a converter declared only for a subtype of it's return type, e.g.:

```
Converter listConv = s -> Arrays.asList(s.split(",");
builder.withConverter(listConv, ArrayList.class);
builder.withConverter(listConv, LinkedList.class);
// but another converter for all other List subtypes (with lower priority)
builder.withConverter(defaultListConv, List.class);
```

A third (optional) argument allowing to specify the priority would be handy too, because it's not possible to annotate lambdas with @Priority either. But I would leave both suggestions to after Config 1.0.

**Comment by Emily-Jiang** [comment]
same as #205

**Comment by Emily-Jiang** [comment]
solved by #298
